feathr_project/feathr/udf/_preprocessing_pyudf_manager.py

```diff
@@ -73,23 +76,25 @@ def build_anchor_preprocessing_metadata(anchor_list: List[FeatureAnchor], local_
         with open(feathr_pyspark_metadata_abs_path, 'wb') as file:
             pickle.dump(features_with_preprocessing, file)
 
+
     @staticmethod
-    def _parse_function_str_for_name(source: str) -> str:
-        """
-        Use AST to parse the functions and get the name out.
+    def _parse_function_str_for_name(fn_str: str) -> str:
+        """Use AST to parse the function string and get the name out.
+
+        Args:
+            fn_str: Function code in string.
+
+        Returns:
+            Name of the function.
         """
-        if source is None:
+        if not fn_str:
             return None
-        tree = ast.parse(source)
+
+        tree = ast.parse(fn_str)
         if len(tree.body) != 1 or not isinstance(tree.body[0], ast.FunctionDef):
-            raise ValueError('provided code fragment is not a single function')
-        code = compile(source=tree, filename='custom.py',mode= 'exec')
-        # https://docs.python.org/3/library/inspect.html see the inspect module for more details
-        # tuple of names other than arguments and function locals. Assume there will be only one function, so will return the first as the name
-        for ele in code.co_consts:
-            # find the first object, that is the str, this will be the name of the function
-            if isinstance(ele, str):
-                return ele
+            raise ValueError("provided code fragment is not a single function")
+
+        return tree.body[0].name
```

Comment:
Thanks! Maybe add a bit explanation in the PR or in the comment on how tree.body[] works?

Reply:
Thank you for the review! tree.body returns the list of `FunctionDef` where if you pass N number of functions in a string, the `len(tree.body) == N`. So tree.body[0] returns the first function definition and `.name` returns its name. (Our expected input fn_str is anyway a single udf, otherwise it throws an exception).

Reply:
Sounds good... I mean put it into the comments in the code for easier maintenance for other devs